crates/bindings-macro/src/lib.rs

```diff
@@ -436,13 +457,61 @@ struct Column<'a> {
     attr: ColumnAttribute,
 }
 
-fn spacetimedb_table(item: TokenStream, public: Option<Span>) -> syn::Result<TokenStream> {
+fn spacetimedb_table(item: TokenStream, public: Option<Span>, scheduled: Option<Ident>) -> syn::Result<TokenStream> {
     let public = public.map(|span| quote_spanned!(span => #[sats(public)]));
 
+    let output = if let Some(reducer) = scheduled {
+        let reducer_name = reducer.to_string();
+        let mut modified_item = syn::parse2::<DeriveInput>(item)?;
+        let type_check = reducer_type_check(&modified_item, &reducer)?;
+        add_scheduled_fields(&mut modified_item)?;
+        quote! {
+            #[derive(spacetimedb::TableType)]
+            #public
+            #[sats(scheduled=#reducer_name)]
+            #modified_item
+            #type_check
+        }
+    } else {
+        quote! {
+            #[derive(spacetimedb::TableType)]
+            #public
+            #item
+        }
+    };
+    Ok(output)
+}
+
+// add scheduled_id and scheduled_at fields to the struct
+fn add_scheduled_fields(item: &mut DeriveInput) -> syn::Result<()> {
+    match &mut item.data {
+        syn::Data::Struct(ref mut struct_data) => match &mut struct_data.fields {
+            syn::Fields::Named(fields) => {
+                fields.named.extend([
+                    syn::Field::parse_named.parse2(quote! {#[autoinc]
+                    pub scheduled_id: u64 })?,
```

Comment:
In the spec this is called `id`, but here it's called `scheduled_id` - which one should be updated?